lib/config_cat/config_fetcher.ex

```diff
@@ -216,9 +221,15 @@ defmodule ConfigCat.CacheControlConfigFetcher do
     response
   end
 
-  defp log_error(error) do
+  defp log_error(error, state) do
     Logger.error("Double-check your SDK Key at https://app.configcat.com/sdkkey.")
     Logger.error("Failed to fetch configuration from ConfigCat: #{inspect(error)}")
+
+    if error == {:error, %HTTPoison.Error{reason: :timeout}} do
```

Comment:
This conditional probably won't work like you expect because it's doing a strict equality comparison, and an `HTTPoison.Error` struct almost certainly has more fields than just the one being checked here.

The earlier version with the `case` statement would have used pattern matching instead of direct equality comparison and that would have worked.

Options:
- You could go back to the `case` version you had earlier
- You could look at using [`Kernel.match?`](https://hexdocs.pm/elixir/1.13/Kernel.html#match?/2) to do the comparison
- You could extract a private helper function, something like:

```elixir
defp log_timeout({:error, %HTTPoison.Error{reason: :timeout}}) do
  Logger.error(...)
end

defp log_timeout(_error), do: :ok
```